optuna/storages/_in_memory.py

```diff
@@ -471,3 +472,106 @@ def __init__(self, name: str) -> None:
         self.name: str = name
         self.directions: List[StudyDirection] = [StudyDirection.NOT_SET]
         self.best_trial_id: Optional[int] = None
+
+
+class _Manager(BaseManager):
+    pass
+
+
+_Manager.register("InMemoryStorage", InMemoryStorage)
+
+
+class MultiprocessInMemoryStorage(BaseStorage):
```

Comment:
This is a very simple implementation 👍

Reply:
[nits] This is not a strong opinion, but `IPCStorage` or `InterProcessStorage` might be better.